firstpartydata/first_party_data.go

```diff
@@ -479,14 +350,17 @@ func resolveApp(fpdConfig *openrtb_ext.ORTB2, bidRequestApp *openrtb2.App, globa
 		}
 	}
 
-	newApp := *bidRequestApp
-	var err error
+	newApp := ortb.CloneApp(bidRequestApp)
```

Comment:
Not sure if the deep clone should be performed here or in the `mergeApp` function.

A shallow clone is made here and that's fine for the global ext and data fields. The ORTB fields require a deep clone. If we do the deep clone here we're wasting cpu if we don't have ORTB data to merge. If we keep with the old shallow clone, we'll need to redo that work later for the ORTB deep clone.

I'm conflicted. Any advice?

Reply:
Nevermind. We need to deep clone the fields applied by other levels of FPD, such as `Content` and `Content.Dats`, before we potentially overwrite it with the ORTB2 merge. Needs to stay as-is.